org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.xtend

```diff
@@ -219,6 +222,14 @@ class RenameService2 implements IRenameService2 {
 		return null
 	}
 
+	protected def String getConvertedValue(EObject grammarElement, ILeafNode leaf) {
+		switch (grammarElement) {
+			RuleCall: valueConverterService.toValue(leaf.text, grammarElement.rule.name, leaf).toString()
```

Comment:
`toValue()` on a leaf node could throw a value converter exception if the leaf node was used in a data type rule e.g.
`IntAsID: INT;` with `999999999999999999`.
The value converter exception is not caught / handled in `doPrepareRename`

Reply:
Would you really expect rename to work OOTB when integers are used as names? 
Anyway, I added some code to handle this.